src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -599,8 +632,17 @@ def _visit_map(self, node: gtir.FunCall) -> DataExpr:
         fun_python_code = gtir_python_codegen.get_source(fun_node)
         tasklet_expression = f"{output_connector} = {fun_python_code}"
 
-        # TODO(edopao): extract offset_dim from the input arguments
-        offset_dim = gtx_common.Dimension("", gtx_common.DimensionKind.LOCAL)
+        input_local_offsets = [
+            input_arg.local_offset for input_arg in input_args if input_arg.local_offset is not None
+        ]
+        if len(input_local_offsets) == 0:
+            raise ValueError(f"Missing information on local dimension for map node {node}.")
+
+        local_offset = input_local_offsets[0]
+        offset_provider = self.subgraph_builder.get_offset_provider(local_offset)
+        assert isinstance(offset_provider, gtx_common.Connectivity)
+        local_size = offset_provider.max_neighbors
+        offset_dim = gtx_common.Dimension(local_offset, gtx_common.DimensionKind.LOCAL)
```

Comment:
Next dump question, but is the offset provider also always the name of local dimension it produces.
Shouldn't it have something like a `Dim` at the end?

Reply:
Then I would add a description about this, because it looks strange.
An alternative would also be to modify `get_map_variable()` to accept different arguments.
I mean it is Python and Python has ducktyping.

Reply:
I will change the signature of the function.